### DISCUSSION TITLE
Way to specify a maximal witdh in wide terminals?

### DISCUSSION THREAD

**Discussion by con-f-use** [fastapi/typer#680] [Questions] [answered]
### Description

When using rich markdown formatting in a wide terminal, I'd find it quite useful to limit the maximal width of bounding boxes (see image). Is there something like a `max_width` option or could I pass `rich.Console(width=200)` somewhere, like with the `RichHandler` for logging? Ideally the bounding boxes would just automatically span the width of the widest line in them.

![image](https://github.com/tiangolo/typer/assets/11145016/2263c303-0380-4b06-b036-2aa95a66c4f0)

### Example Code

```python
#!/usr/bin/env python

import logging, typer
from rich.logging import RichHandler, Console

app = typer.Typer(rich_markup_mode="markdown")

@app.callback()
def main( ctx: typer.Context):
    logging.basicConfig(
        handlers=[RichHandler(console=Console(width=120))],
    )

if __name__ == "__main__":
    app()
```

<details>

  <summary>Details (click here)</summary>

  ### Operating System

  Linux

  ### Operating System Details

  _No response_

  ### Typer Version

  0.9.0

  ### Python Version

  3.10

  ### First Check

  - [X] I added a very descriptive title here.
  - [X] I used the GitHub search to find a similar question and didn't find it.
  - [X] I searched the Typer documentation, with the integrated search.
  - [X] I already searched in Google "How to X in Typer" and didn't find any information.
  - [X] I already read and followed all the tutorial in the docs and didn't find an answer.
  - [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

  ### Commit to Help

  - [X] I commit to help with one of the help options

</details>

**Comment by Apocryphon-X** [comment]
> **(Unanswered)** Related Discussion: https://github.com/tiangolo/typer/discussions/607

**Comment by con-f-use** [comment]
Thanks, that works for error boxes, but doesn't seem to work for everything:

![image](https://github.com/tiangolo/typer/assets/11145016/2a27ad23-98c3-4a71-91d9-60aa7280c342)

```python
#!/usr/bin/env python

import logging, typer
from rich.logging import RichHandler, Console
from rich.panel import Panel

typer.rich_utils.Panel = Panel.fit

app = typer.Typer(rich_markup_mode="markdown")

@app.callback()
def main( ctx: typer.Context):
    logging.basicConfig(
        handlers=[RichHandler(console=Console(width=120))],
    )

if __name__ == "__main__":
    app()
```

**Comment by con-f-use** [comment]
Closing this as duplicate of https://github.com/tiangolo/typer/discussions/607

**Comment by svlandeg** [comment, marked as answer]
cf https://github.com/fastapi/typer/pull/528#issuecomment-3311701583: you can set a `TERMINAL_WIDTH` env. var.
